types/hooks.d.ts

```diff
@@ -0,0 +1,235 @@
+/// <reference types="./react-spring"/>
+
+declare module 'react-spring/hooks' {
+  import {
+    SpringConfig,
+    SpringBaseProps,
+    TransitionKeyProps,
+  } from 'react-spring'
+
+  /** List from `function getForwardProps` in `src/shared/helpers` */
+  type ExcludedProps =
+    | 'to'
+    | 'from'
+    | 'config'
+    | 'native'
+    | 'onStart'
+    | 'onRest'
+    | 'onFrame'
+    | 'children'
+    | 'reset'
+    | 'reverse'
+    | 'force'
+    | 'immediate'
+    | 'impl'
+    | 'inject'
+    | 'delay'
+    | 'attach'
+    | 'destroyed'
+    | 'track'
+    | 'interpolateTo'
+    | 'autoStart'
+    | 'ref'
+  type ForwardedProps<T> = Pick<T, Exclude<keyof T, ExcludedProps>>
+  // NOTE: because of the Partial, this makes a weak type, which can have excess props
+  type InferFrom<T extends object> = T extends { to: infer TTo }
+    ? Partial<TTo>
+    : Partial<ForwardedProps<T>>
```

Comment:
`interpolateTo`'s fault 😭

This is likely why the current types for `Spring`, `Trail` et al have a required `to` prop. This is my attempt to make the hooks not have the `to` prop required.